modules/objdetect/src/aruco/aruco_board.cpp

```diff
@@ -368,25 +367,29 @@ void CharucoBoardImpl::calcNearestMarkerCorners() {
     }
 }
 
-void CharucoBoardImpl::matchImagePoints(InputArrayOfArrays detectedCorners, InputArray detectedIds,
-                                   OutputArray _objPoints, OutputArray imgPoints) const {
-    if (detectedCorners.kind() == _InputArray::STD_VECTOR_VECTOR ||
-        detectedCorners.isMatVector() || detectedCorners.isUMatVector())
-        Board::Impl::matchImagePoints(detectedCorners, detectedIds, _objPoints, imgPoints);
-    else {
-        CV_Assert(detectedCorners.isMat() || detectedCorners.isVector());
-        size_t nDetected = detectedCorners.total();
-        vector<Point3f> objPnts(nDetected);
-        vector<Point2f> imgPnts(nDetected);
-        for(size_t i = 0ull; i < nDetected; i++) {
-            int pointId = detectedIds.getMat().at<int>((int)i);
-            CV_Assert(pointId >= 0 && pointId < (int)chessboardCorners.size());
-            objPnts[i] = chessboardCorners[pointId];
-            imgPnts[i] = detectedCorners.getMat().at<Point2f>((int)i);
+void CharucoBoardImpl::matchImagePoints(InputArrayOfArrays detectedCharuco, InputArray detectedIds,
+                                        OutputArray outObjPoints, OutputArray outImgPoints) const {
+    CV_Assert(detectedIds.total() == detectedCharuco.total());
+    CV_Assert(detectedIds.total() > 0ull);
+    // detectedCharuco includes charuco corners as vector<Point2f> or Mat.
+    // Python bindings could add extra dimension to detectedCharuco and therefore vector<Mat> case is additionally rocessed.
```

Comment:
fixed typo

Reply:
I suggest @asmorkalov to create new `CharucoBoard::matchImagePoints` with **InputArray** signature:

```
    CV_WRAP void matchImagePoints(InputArray detectedCharuco, InputArray detectedIds,
                                  OutputArray objPoints, OutputArray imgPoints) const;
```

But in this case, two Python functions are generated:

- matchImagePoints(**InputArray** detectedCharuco, ...)
- matchImagePoints(**InputArrayOfArrays** detectedCharuco, ...)

To avoid a potential problem @asmorkalov suggest use only default `Board::matchImagePoints` method

I don't know which way is better.